cloudferrylib/scheduler/scenario.py

```diff
@@ -176,3 +194,27 @@ def __init__(self, scenario_checker):
         self.duplicated_tasks = scenario_checker.duplicated_tasks
         self.missed_links = scenario_checker.missed_links
         super(ScenarioError, self).__init__("Not valid scenario")
+
+
+class IncorrectScenarioStructure(RuntimeError):
+    def __init__(self):
+        super(IncorrectScenarioStructure, self)\
+            .__init__("Empty or not dictonary-like strcuture")
+
+
+class NoStagesScenario(RuntimeError):
+    def __init__(self):
+        super(NoStagesScenario, self)\
+            .__init__("None of any of the basic steps for migration")
+
+
+class NotFoundScenarioFileError(RuntimeError):
+    def __init__(self, path):
+        super(NotFoundScenarioFileError, self).__init__(
+            "Not found on path %s" % path)
+
+
+class NotFoundTasksFileError(RuntimeError):
+    def __init__(self, path):
+        super(NotFoundTasksFileError, self).__init__(
+            "Not found on path %s" % path)
```

Comment:
None of these exceptions provide enough information to the user, neither does the calling code. What I'd suggest is for  incorrect scenario to provide the file and line number where the error was spotted, with some sort of clue to how to resolve the problem. For the "file not found" types of errors you should provide more information, like "Make sure you specified correct config value <config option> in <config file name>".

The other thing to improve is to have some better structure for exceptions and handling those appropriately, for example, "invalid scenario" exception is most likely a syntax error, while "file not found" exceptions are "invalid configuration" errors, those need to be caught and processed appropriately in the calling code.